app/models/setting/discovered.rb

```diff
@@ -28,6 +28,8 @@ def self.load_defaults
         self.set('discovery_facts_hardware', N_("Regex to organize facts for hardware section"), "", N_("Hardware facts")),
         self.set('discovery_facts_network', N_("Regex to organize facts for network section"), "", N_("Network facts")),
         self.set('discovery_facts_ipmi', N_("Regex to organize facts for ipmi section"), "", N_("IPMI facts")),
+        self.set('discovery_lock', N_("Automatically generate PXE configuration to pin a newly discovered host to discovery"), false,N_("Lock PXE")),
+        self.set('discovery_lock_template', N_("PXE template to be used when pinning a host to discovery"), 'pxelinux_discovery',N_("PXE lock template name")),
```

Comment:
> @lzap not sure I am following you on the Proc - what changed?

Please change the syntax as in the example provided, for performance
reasons (perhaps) the API expects a proc instead of hash.

-- 
Later,
 Lukas #lzap Zapletal